--- gcc/cp/g++spec.cc	2025-10-24 18:38:46.452718757 -0700
+++ gcc/cp/g++spec.cc	2025-10-24 18:39:00.565987367 -0700
@@ -97,7 +97,7 @@
   int library = 0;
 
   /* Which c++ runtime library to link.  */
-  stdcxxlib_kind which_library = USE_LIBSTDCXX;
+  stdcxxlib_kind which_library = USE_LIBCXX;
 
   /* The number of arguments being added to what's in argv, other than
      libraries.  We use this to track the number of times we've inserted

--- gcc/c-family/c.opt	2025-10-30 13:22:21.184157614 -0700
+++ gcc/c-family/c.opt	2025-10-30 13:21:57.802662053 -0700
@@ -2791,7 +2791,7 @@
 Conform to the ISO 2023 C standard (published in 2024).
 
 stdlib=
-Driver C++ ObjC++ Common Condition(ENABLE_STDLIB_OPTION) Var(flag_stdlib_kind) Joined Enum(stdlib_kind) RejectNegative Init(1)
+Driver C++ ObjC++ Common Condition(ENABLE_STDLIB_OPTION) Var(flag_stdlib_kind) Joined Enum(stdlib_kind) RejectNegative Init(2)
 -stdlib=[libstdc++|libc++]	The standard library to be used for C++ headers
 and runtime.
 
